tests/TestCase/ORM/AssociationTest.php

```diff
@@ -170,6 +171,43 @@ public function testTarget()
     }
 
     /**
+     * Tests that target() returns the correct Table object for plugins
+     *
+     * @return void
+     */
+    public function testTargetPlugin()
+    {
+        Plugin::load('TestPlugin');
+
+        $config = [
+            'className' => 'TestPlugin.Comments',
+            'foreignKey' => 'a_key',
+            'conditions' => ['field' => 'value'],
+            'dependent' => true,
+            'sourceTable' => $this->source,
+            'joinType' => 'INNER'
+        ];
+        $this->association = $this->getMock(
+            '\Cake\ORM\Association',
+            [
+                '_options', 'attachTo', '_joinCondition', 'cascadeDelete', 'isOwningSide',
+                'saveAssociated', 'eagerLoader', 'type'
+            ],
+            ['ThisAssociationName', $config]
```

Comment:
[Association is an abstract class](https://github.com/cakephp/cakephp/blob/3.0/src/ORM/Association.php#L31) - so it can't be instanciated directly, rather than create an empty double, I used a mock.

The mocked methods are [copied from the startup method](https://github.com/cakephp/cakephp/blob/3.0-table-registry-alias/tests/TestCase/ORM/AssociationTest.php#L65-L72) - which I should get rid of though.